definitions/npm/react-router-dom_v6.x.x/flow_v0.201.x-/react-router-dom_v6.x.x.js

```diff
@@ -226,6 +226,246 @@ declare module 'react-router-dom' {
 
   declare export type InitialEntry = string | Partial<Location>;
 
+  declare type LowerCaseFormMethod =
+    | 'get'
+    | 'post'
+    | 'put'
+    | 'patch'
+    | 'delete';
+  declare type UpperCaseFormMethod =
+    | 'GET'
+    | 'POST'
+    | 'PUT'
+    | 'PATCH'
+    | 'DELETE';
+
+  /**
+   * Users can specify either lowercase or uppercase form methods on <Form>,
+   * useSubmit(), <fetcher.Form>, etc.
+   */
+  declare export type HTMLFormMethod =
+    | LowerCaseFormMethod
+    | UpperCaseFormMethod;
+
+  declare type V7_FormMethod = UpperCaseFormMethod;
+
+  declare export type FormEncType =
+    | 'application/x-www-form-urlencoded'
+    | 'multipart/form-data';
+
+  /**
+   * Potential states for state.navigation
+   */
+  declare export type NavigationStates = {
+    Idle: {
+      state: 'idle',
+      location: void,
+      formMethod: void,
+      formAction: void,
+      formEncType: void,
+      formData: void,
+    },
+    Loading: {
+      state: 'loading',
+      location: Location,
+      formMethod: RemixFormMethod | V7_FormMethod | void,
```

Comment:
I realise these are copied from TS and they generally do this from what I've noticed, as a nit I think the following is easier, but not a requirement to change
```suggestion
      formMethod?: RemixFormMethod | V7_FormMethod,
```

Reply:
oh: I wrongly supposed that `{ formMethod: undefined }` won't work but it does since a while.

Reply:
needed cf. comment above